tests/TypeReconciliation/ReconcilerTest.php

```diff
@@ -177,4 +176,61 @@ public function providerTestTypeIsContainedBy(): array
             ],
         ];
     }
+
+    /**
+     * @dataProvider constantAssertions
+     */
+    public function testReconciliationOfClassConstantInAssertions(string $assertion, string $expected_type): void
```

Comment:
Is there a more solid way to test this in a more "integrated" way?
This wont ensure that `class-constant(Foo::*)` will be passed to the `AssertionReconciler#reconcile` and thus it may break when something regarding this will change in a later version.

Reply:
Not sure I understood what you meant there

Reply:
In this test, I have to assume that the caller of the class-function has already made its job.
Given the fact that wildcard constants worked in a previous version (I've used that in a project), I think there was no proper test. I think the unit tests are still passing even tho the functionality is gone.